tests/buildenv-fedora.dockerfile

```diff
@@ -88,4 +88,23 @@ COPY --from=alpine-lib-image $PREFIX $PREFIX/
 # take libcrypto from Fedora - python ssl doesn't like alpine one
 RUN cp /usr/lib64/libcrypto.so.1.1.1[a-z] ${PREFIX}/alpine-lib/lib/libcrypto.so.1.1
 
+# nix install will use sudo. Allow it do happen with no passwd. we'll turn this off
+# after nix is installed.
+RUN echo "%appuser ALL=(ALL) NOPASSWD: ALL" > /etc/sudoers.d/XX-for-nix-install
+
 USER $USER
+
+# Install nix package management in the final container.
+RUN curl -L -o install_nix.sh https://nixos.org/nix/install && \
+    chmod +x install_nix.sh && \
+    ./install_nix.sh --daemon && \
```

Comment:
why is sudo required down below but not here?

Reply:
The `install_nix.sh` script complains mightily of it is run as root, but the script wants to use `sudo`. Hence the dance around `/etc/sudoers.d` and `USER`.